dwi_ml/experiment/trainor_abstract.py

```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/env python
+
+import datetime
+import json
+import os
+
+import time
+
+import numpy as np
+import torch
+
+from dwi_ml.data.dataset.data_list import \
+    (MultiSubjectDataset, LazyMultiSubjectDataset)
+
+# À ARRANGER
+from scil_vital.shared.code.utils.timer import Timer
+from scil_vital.shared.code.utils.experiment_utils import ValueMonitor
+
+
+class TractoMLAbstractLocal:
+    def __init__(self):
+        raise NotImplementedError
+
+    def build_model(self):
+        raise NotImplementedError
+
+    def train(self, **kwargs):
+        raise NotImplementedError
+
+    def save(self):
+        raise NotImplementedError
+
+    def load_model(self, filepath, **kwargs):
+        raise NotImplementedError
+
+
+class TractoMLAbstractSequences:
+    def __init__(self,
+                 train_database_path: str = None,
+                 valid_database_path: str = None,
+                 name: str = None,
+                 # Concerning the choice of inputs:
+                 nb_degree_angles: int = 128,
+                 add_streamline_noise: bool = False,
+                 streamlines_cut_ratio: float = None, step_size: float = None,
+                 neighborhood_dist_mm: float = None,
+                 nb_neighborhood_axes: int = 6,
+                 add_previous_dir: bool = False,
+                 lazy: bool = False,
+                 # Concerning the memory usage:
+                 batch_size: int = 20000, volumes_per_batch: int = None,
+                 cycles_per_volume_batch: int = 1,
+                 n_epoch: int = 100, seed: int = 1234, patience: int = 20,
+                 use_gpu: bool = True, num_workers: int = 0,
+                 worker_interpolation: bool = False,
+                 cache_manager: bool = False, taskman_managed: bool = False):
+        """
+        Mandatory parameters:
+        ---------------------
+        train_database_path : str
+            Path to training database (hdf5 file)
+        valid_database_path : str
+            Path to validation database (hdf5 file)
+
+        Optional parameters:
+        --------------------
+        ====> General
+        name : str
+            Optional name of the experiment. If given, it is prepended to the
+            auto-generated name. [None]
+
+        ====> Concerning the choice of inputs:
+        nb_degree_angles: int
```

Comment:
Will we choose to use `nb` as a short name for number or shall we use `num`? I personally prefer the latter.

Specifically for this argument, the docstring is not clear to me: if it is the number of directions, I would use num_directions, since I would tend to think of an angular resolution is degrees (which can be a float).

Reply:
I think I prefer nb. Anyway we will have to see for all parameters if they really are sharable for all our projects.